x/evm/keeper/msg_server_user_smart_contract.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/palomachain/paloma/x/evm/types"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func (k Keeper) UploadUserSmartContract(
```

Comment:
FYI, since this is the `msgServer` implementation, you can use `msgServer` instead of `Keeper` as a receiver. Take a look at other modules implementation for this. Even though it's the same struct ultimately, it's a bit more explicit about intention.